tests/staticfiles_tests/cases.py

```diff
@@ -0,0 +1,346 @@
+        relpath = self.hashed_file_path("cached/css/window.css")
+        self.assertEqual(relpath, "cached/css/window.3906afbb5a17.css")
+        with storage.staticfiles_storage.open(relpath) as relfile:
+            content = relfile.read()
+            self.assertNotIn(b'url(img/window.png)', content)
+            self.assertIn(b'url("img/window.acae32e4532b.png")', content)
+
+    def test_template_tag_url(self):
+        relpath = self.hashed_file_path("cached/url.css")
+        self.assertEqual(relpath, "cached/url.902310b73412.css")
+        with storage.staticfiles_storage.open(relpath) as relfile:
+            self.assertIn(b"https://", relfile.read())
+
+    def test_post_processing(self):
+        """Test that post_processing behaves correctly.
+
+        Files that are alterable should always be post-processed; files that
+        aren't should be skipped.
+
+        collectstatic has already been called once in setUp() for this testcase,
+        therefore we check by verifying behavior on a second run.
+        """
+        collectstatic_args = {
+            'interactive': False,
+            'verbosity': 0,
+            'link': False,
+            'clear': False,
+            'dry_run': False,
+            'post_process': True,
+            'use_default_ignore_patterns': True,
+            'ignore_patterns': ['*.ignoreme'],
+        }
+
+        collectstatic_cmd = CollectstaticCommand()
+        collectstatic_cmd.set_options(**collectstatic_args)
+        stats = collectstatic_cmd.collect()
+        self.assertIn(os.path.join('cached', 'css', 'window.css'), stats['post_processed'])
+        self.assertIn(os.path.join('cached', 'css', 'img', 'window.png'), stats['unmodified'])
+        self.assertIn(os.path.join('test', 'nonascii.css'), stats['post_processed'])
+
+    def test_css_import_case_insensitive(self):
+        relpath = self.hashed_file_path("cached/styles_insensitive.css")
+        self.assertEqual(relpath, "cached/styles_insensitive.c609562b6d3c.css")
+        with storage.staticfiles_storage.open(relpath) as relfile:
+            content = relfile.read()
+            self.assertNotIn(b"cached/other.css", content)
+            self.assertIn(b"other.d41d8cd98f00.css", content)
+
+    @override_settings(
+        STATICFILES_DIRS=[os.path.join(TEST_ROOT, 'project', 'faulty')],
+        STATICFILES_FINDERS=['django.contrib.staticfiles.finders.FileSystemFinder'],
+    )
+    def test_post_processing_failure(self):
+        """
+        Test that post_processing indicates the origin of the error when it
+        fails. Regression test for #18986.
+        """
+        finders.get_finder.cache_clear()
+        err = six.StringIO()
+        with self.assertRaises(Exception):
+            call_command('collectstatic', interactive=False, verbosity=0, stderr=err)
+        self.assertEqual("Post-processing 'faulty.css' failed!\n\n", err.getvalue())
+
+
+class FinderTestCase(object):
```

Comment:
Should I do the same with `TestHashedFiles` (only used in `test_storage.py`) and `TestNoFilesCreated` (only used in `test_management.py`)?